ui/src/app/utils/cluster-initializer.spec.tsx

```diff
@@ -0,0 +1,305 @@
+    // https://github.com/facebook/jest/issues/2684). So we just set a very quick timeout here to
+    // ensure the threshold is reached after a couple polling loops.
+    const initializer = createInitializer({overallTimeout: 30});
+
+    // Tell Jest that we plan to have 1 assertion. This ensures that the test won't
+    // pass if the promise fails.
+    expect.assertions(1);
+    return runInitializerAndTimers(initializer).catch(error => {
+      expect(error.message).toMatch(/max time allowed/i);
+    });
+  });
+
+  it('Should reject promise after abort signal', async() => {
+    mockGetCluster.mockImplementation(async(workspaceNamespace) => {
+      return {...baseCluster, status: ClusterStatus.Starting};
+    });
+    const aborter = new AbortController();
+    const initializer = createInitializer({
+      abortSignal: aborter.signal
+    });
+
+    const initializePromise = runInitializerAndTimers(initializer);
+
+    // Wait a reasonably-short amount of time, at least one polling delay period, before sending
+    // an abort signal.
+    await new Promise(resolve => setTimeout(resolve, 20));
+    aborter.abort();
+
+    expect.assertions(1);
+    return initializePromise.catch(error => {
+      expect(error.message).toMatch(/aborted/i);
+    });
+  });
+
+  it('Should respect the maxDeleteCount option', async() => {
+    // Mock out getCluster API responses which simulate a cluster in an error state, which is then
+    // reset, but ends up in an error state again. This scenario should warrant two deleteCluster
+    // calls, but the initializer is configured to max out at 1 and should return an error.
+    mockGetClusterCalls([
+      {...baseCluster, status: ClusterStatus.Error},
+      {...baseCluster, status: ClusterStatus.Deleting},
+      {...baseCluster, status: ClusterStatus.Creating},
+      {...baseCluster, status: ClusterStatus.Error},
+    ]);
+    mockDeleteCluster.mockImplementation(async(workspaceNamespace) => {
+      return {};
+    });
+    mockCreateCluster.mockImplementation(async(workspaceNamespace) => {
+      return {...baseCluster, status: ClusterStatus.Creating};
+    });
+
+    const initializer = createInitializer({maxDeleteCount: 1});
+
+    expect.assertions(2);
+    return runInitializerAndTimers(initializer).catch(error => {
+      expect(mockDeleteCluster).toHaveBeenCalledTimes(1);
+      expect(error.message).toMatch(/max cluster delete count/i);
+    });
+  });
+
+  it('Should respect the maxCreateCount option', async() => {
+    // Ensure that the initializer won't take action on a NOT_FOUND cluster if the maxCreateCount
+    // is set to disallow create requests.
+    mockGetCluster.mockImplementationOnce(async(workspaceNamespace) => {
+      throw new Response(null, {status: 404});
+    });
+    const initializer = createInitializer({maxCreateCount: 0});
+    return runInitializerAndTimers(initializer).catch(error => {
```

Comment:
Hmm, I don't see where return values of the `it` statements are checked. It actually looks like the return type for `it` is `void`.